packages/labs/virtualizer/README.md

```diff
@@ -0,0 +1,208 @@
+
+### Flow layout
+
+By default, a virtualizer lays out its children using the `flow` layout, a simplified form of the browser's own default layout.
+
+The `flow` layout's primary (and significant) simplification is that it expects all child elements to be styled as block-level elements and lays them out accordingly. Child elements will never be laid out "next to each other" inline, even if there were enough space to do so.
+
+Child element size is determined "naturally"—that is, the size of each child element will depend on the data you provide in the `items` array, the nature of your `renderItem` template, and any CSS rules that apply to the child.
+
+#### Spacing child elements
+
+To control the spacing of child elements, use standard CSS techniques to set margins on the elements.
+
+Note that the `flow` layout offers limited support for [margin-collapsing](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Box_Model/Mastering_margin_collapsing): margins set explicitly on child elements will be collapsed, but any margins on elements contained _within_ child elements are not considered.
+
+#### Specifying layout direction
+
+The `flow` layout works vertically by default. However, it also supports laying out child elements horizontally, via its `direction` property:
+
+```js
+  render() {
+    return html`
+      <lit-virtualizer
+        .layout=${flow({
+          direction: 'horizontal'
+        })}
+        .items=${this.photos}
+        .renderItem=${photos => html`<img src=${photo.url}>`}
+      ></lit-virtualizer>
+    `;
+  }
+
+```
+
+### Grid layout
+
+TODO
+
+### Scrolling to a specific item or position
+
+TODO
+
+### `virtualize` directive
+
+An alternative way to use `@lit-labs/virtualizer` is via the `virtualize` directive; this directive turns its parent element into a virtualizer.
+
+The `virtualize` directive is useful primarily if your project utilizes Lit templates but not the `LitElement` base class and you want to keep your imports to a minimum.
+
+It can also be useful in cases like the one below, where we want to virtualize an unordered list (`<ul>`) element and don't want or need a separate container for its `<li>` children:
+
+```js
+render() {
+  return html`
+    <ul>
+      ${virtualize({
+        items: this.contacts,
+        renderItem: contact => html`<li>${contact.name}</li>`
+      })}
+    </ul>
+  `;
+}
+```
+
+The capabilities of the `virtualizer` directive are the same as those of the `<lit-virtualizer>` element. The APIs are the same as well, except that features expressed as properties and attributes on the `<lit-virtualizer>` element are instead expressed as properties in an options object passed as the single argument to the `virtualize` directive.
+
+## API Referrence
```

Comment:
We should wire this package into the generated API docs. We probably need to do that for a number of packages. File an issue? cc @aomarks

Reply:
Filed an issue in this repo #3037